### PR TITLE
perf: reduce chat responses fallback usage allocation

### PR DESCRIPTION
**Pull request by jstar0**
## 📝 变更描述 / Description

This PR reduces allocation pressure in the OpenAI-compatible chat stream path when a request is internally routed through Responses and has to fall back to local usage counting.

On current `main`, Responses stream conversion is owned by `relaykit/relayconvert`. The converter previously retained all reasoning, output-text, tool-name, and tool-argument chunks in a `strings.Builder`; the handler then converted that builder to a complete string for `ResponseText2Usage`. Long streams therefore kept a duplicate response-sized buffer alive until completion.

The updated converter accepts an optional usage-text sink. The chat-via-Responses handler supplies an incremental estimator, so the converter forwards the exact same chunk sequence without retaining the complete usage text. Existing relaykit consumers that do not provide a sink keep the original `UsageText()` behavior.

The fallback trigger is unchanged: valid upstream usage still takes precedence, and local-count context is set only when upstream usage is absent or empty. Exact current head: `5dafa9f343b537a21c34ba9a3eaaf605b2a2c340`.

## 🚀 变更类型 / Type of change

- [ ] 🐛 Bug 修复 (Bug fix) - *请关联对应 Issue，避免将设计取舍、理解偏差或预期不一致直接归类为 bug*
- [ ] ✨ 新功能 (New feature) - *重大特性建议先通过 Issue 沟通*
- [x] ⚡ 性能优化 / 重构 (Refactor)
- [ ] 📝 文档更新 (Documentation)

## 🔗 关联任务 / Related Issue

- Related to #5480

## ✅ 提交前检查项 / Checklist

- [x] **人工确认:** 我已亲自整理并撰写此描述，没有直接粘贴未经处理的 AI 输出。
- [x] **非重复提交:** 我已搜索现有的 [Issues](https://github.com/QuantumNous/new-api/issues) 与 [PRs](https://github.com/QuantumNous/new-api/pulls)，确认不是重复提交。
- [x] **Bug fix 说明:** 若此 PR 标记为 `Bug fix`，我已提交或关联对应 Issue，且不会将设计取舍、预期不一致或理解偏差直接归类为 bug。
- [x] **变更理解:** 我已理解这些更改的工作原理及可能影响。
- [x] **范围聚焦:** 本 PR 未包含任何与当前任务无关的代码改动。
- [x] **本地验证:** 已在本地运行并通过测试或手动验证，维护者可以据此复核结果。
- [x] **安全合规:** 代码中无敏感凭据，且符合项目代码规范。

## 📸 运行证明 / Proof of Work

Current-head tests and independent relaykit build:

```text
$ go test ./service ./relay/channel/openai -count=1
ok  github.com/QuantumNous/new-api/service
ok  github.com/QuantumNous/new-api/relay/channel/openai

$ go test ./relay/common ./relay/channel/openai ./relay/channel/codex ./relay/common_handler ./service -count=1
ok  github.com/QuantumNous/new-api/relay/common
ok  github.com/QuantumNous/new-api/relay/channel/openai
ok  github.com/QuantumNous/new-api/relay/channel/codex
?   github.com/QuantumNous/new-api/relay/common_handler [no test files]
ok  github.com/QuantumNous/new-api/service

$ cd relaykit && GOWORK=off go test ./relayconvert/... -count=1
ok  github.com/QuantumNous/new-api/relaykit/relayconvert
ok  github.com/QuantumNous/new-api/relaykit/relayconvert/internal/oai_chat
ok  github.com/QuantumNous/new-api/relaykit/relayconvert/internal/oai_responses
ok  github.com/QuantumNous/new-api/relaykit/relayconvert/kitutil

$ cd relaykit && GOWORK=off go build ./...
PASS

$ git diff --check origin/main..HEAD
PASS
```

The tests cover chunk-boundary equivalence for OpenAI, Claude, and Gemini multipliers; byte-split UTF-8 and invalid trailing bytes; reasoning-summary separators; visible output; tool names and arguments; upstream-usage priority; computed upstream totals; default retained `UsageText()` behavior; and direct plus multi-hop sink behavior.

Ten-sample benchmark on local darwin/arm64 (Apple M4 Pro):

```text
$ go test ./service -run '^$' -bench '^BenchmarkChatViaResponsesFallbackUsage$' -benchmem -count=10
old_builder_openai_large     284281-284282 B/op  21 allocs/op
new_streaming_openai_large              353 B/op   2 allocs/op
old_builder_claude_large     284281-284282 B/op  21 allocs/op
new_streaming_claude_large              353 B/op   2 allocs/op
PASS
```

## Summary by CodeRabbit

* **New Features**
  * Improved token usage reporting for streamed responses, including reasoning, output, and tool-call content.
  * Added model-aware streaming token estimation when usage data is unavailable or incomplete.
  * Preserved detailed upstream usage information when valid data is provided.

* **Bug Fixes**
  * Fixed incomplete usage totals for streams reporting zero or missing token counts.
  * Improved handling of split and partial UTF-8 content during estimation.

* **Tests**
  * Added coverage for fallback usage calculation, tool calls, upstream usage, and streaming edge cases.

### CODE REVIEW

File: relay/channel/openai/chat_via_responses_test.go [hunk 1]
```diff
@@ -39,6 +43,101 @@ func newResponsesChatTestContext(t *testing.T, body string, isStream bool) (*gin
...
+	c, recorder, resp, info := newResponsesChatTestContext(t, body, true)
+	usage, err := OaiResponsesToChatStreamHandler(c, info, resp)
+	require.Nil(t, err)
+	require.NotNil(t, usage)
+	assert.Equal(t, 11, usage.PromptTokens)
+	assert.Equal(t, 13, usage.CompletionTokens)
+	assert.Equal(t, 24, usage.TotalTokens)
+	assert.Equal(t, 11, usage.InputTokens)
+	assert.Equal(t, 13, usage.OutputTokens)
+	assert.False(t, common.GetContextKeyBool(c, constant.ContextKeyLocalCountTokens))
```
[the remark this answers is not in the document]
Comment (jstar0) [reply relay/channel/openai/chat_via_responses_test.go:137]: Thanks for checking. This test intentionally covers normalization of partial upstream usage, not the local fallback. When upstream total_tokens is zero but input_tokens and output_tokens are present, responsesUsageToChatUsage sets TotalTokens to PromptTokens + CompletionTokens in relaykit/relayconvert/internal/oai_responses/to_oai_chat_resp.go. The handler therefore receives 11/13/24 and does not enter the TotalTokens == 0 fallback; the focused test passes and confirms the local-count flag remains false. TestOaiResponsesToChatStreamFallbackUsageMatchesResponseText2Usage separately covers the all-zero fallback. No code change is needed here.
